### PR TITLE
Move imshow closer to px pattern

### PR DESCRIPTION
**Pull request by nicolaskruchten**
@emmanuelle here's the first pass on the PX-ification of the `imshow` function. I'm realizing that to align better with the other `px` methods we should probably add `height`, `width` and `template` and the rest of the `px` defaulting machinery...

### CODE REVIEW

**Comment by nicolaskruchten** [comment]
Here I've bolted on the applicable parts of the "defaults cascade" logic for which I added a test suite the other day :)

**Comment by nicolaskruchten** [comment]
This lets us do stuff like this:

image

File: packages/python/plotly/plotly/express/_imshow.py [hunk 1]
```diff
@@ -74,16 +88,38 @@ def imshow(img, zmin=None, zmax=None, origin=None, colorscale=None):
...
         with a 5% tolerance. For a single-channel image, the max of the image is used.
 
     origin : str, 'upper' or 'lower' (default 'upper')
         position of the [0, 0] pixel of the image array, in the upper left or lower left
         corner. The convention 'upper' is typically used for matrices and images.
 
-    colorscale : str
-        colormap used to map scalar data to colors (for a 2D image). This parameter is not used for
-        RGB or RGBA images.
+    color_continuous_scale : str or list of str
```
Comment (emmanuelle) [inline packages/python/plotly/plotly/express/_imshow.py]: `str (name of colorscale) or list of str (list of css colors)` would be clearer here.

Comment (emmanuelle) [reply packages/python/plotly/plotly/express/_imshow.py]: ok it's explained below, it's fine

**Comment by emmanuelle** [comment]
:dancer: when the CI is green. Which reminds me that we should add imshow examples to the percy test file, one with single-channel and one with RGB. If you don't want to add it here (I should have done this in the initial PR) I'll make another PR for this.

**Comment by nicolaskruchten** [comment]
Ah right yes... If you could do it that would be great :)

**Comment by nicolaskruchten** [comment]
Ah drat also we need to tweak the PR in py-docs

**Pull request merged**